### ISSUE TITLE
Generate request/response structs from OpenAPI spec

### ISSUE THREAD

**Issue by kishorenc** [typesense/typesense-rust#19]
## Description

We've a more or less recent Open API spec that was created as part of the Go client implementation here: https://github.com/typesense/typesense-go/blob/master/typesense/api/openapi.yml

I think it would be a good idea to generate the request/response classes/structs using an Open API client generator like Swagger. This would save a lot of time and ensure that we can keep them up-to-date easily across various client implementations.

What do you think @morenol @emmanuelantony2000 ?

**Comment by emmanuelantony2000** [comment]
Yeah I would support this. Only thing blocking for me is, that we are using a custom transport implementation, which would work with wasm and hyper. AFAIK, Swagger codegen supports only a hyper impl. Correct me if I am wrong.
